test/androidTest/java/org/thoughtcrime/securesms/PinnedMessageEspressoTest.java

```diff
@@ -129,6 +129,82 @@ public void canPinImageMms() {
                 .assertId(R.id.thumbnail_image);
     }
 
+    @Test
+    public void canPinAudio() {
+        Helper helper = new Helper(mainActivityRule);
+
+        String testString = helper.randString();
+
+        helper
+            .goConversations()
+            .goConversation()
+                .sendAudio()
+                .pinMessage(0)
+                .selectMessage(0)
+                .assertId(R.id.menu_context_unpin_message);
```

Comment:
are you sure this is the best way to test this? can you go into the pinned message view and check that the message with that ID is there?

Reply:
same question for all the tests below as well.

Reply:
I'm not sure we have access to the message IDs with espresso. This "technique" was created when I was there and I remember us having trouble finding an alternative solution. We considered checking for the existence of the pinned icon in the message, but it isn't much better.